contracts/utils/contracts/src/ReentrancyGuard.sol

```diff
@@ -25,25 +25,33 @@ import "./LibRichErrors.sol";
 contract ReentrancyGuard {
 
     // Locked state of mutex.
-    bool private locked = false;
+    bool private _locked = false;
 
     /// @dev Functions with this modifer cannot be reentered. The mutex will be locked
     ///      before function execution and unlocked after.
     modifier nonReentrant() {
+        _lockMutexOrThrowIfAlreadyLocked();
+        _;
+        _unlockMutex();
+    }
+
+    function _lockMutexOrThrowIfAlreadyLocked()
```

Comment:
Out of scope: It would be nice if we had a library for automagically spinning up new mutexes on an as needed basis.